lib/matplotlib/axis.py

```diff
@@ -1568,6 +1568,9 @@ def set_major_formatter(self, formatter):
 
         ACCEPTS: A :class:`~matplotlib.ticker.Formatter` instance
         """
+        if not hasattr(formatter, 'format_data'):
+            raise TypeError("formatter argument should be instance of "
```

Comment:
one could shorten this to 
~~~
raise TypeError("formatter must be a matplotlib.ticker.Formatter")
~~~
But that's a matter of taste.

Reply:
Thats what I had before but folks wanted "instance of" :wink: